// react-native-esc-pos-printer/lib/module/printer/constants/enums/barcode.js
"use strict";

import { EscPosPrinter } from "../../../specs/index.js";
const EscPosPrinterConstants = EscPosPrinter.getConstants();
export let BarcodeType = function (BarcodeType) {
  BarcodeType[BarcodeType["BARCODE_UPC_A"] = EscPosPrinterConstants.BARCODE_UPC_A] = "BARCODE_UPC_A";
  BarcodeType[BarcodeType["BARCODE_UPC_E"] = EscPosPrinterConstants.BARCODE_UPC_E] = "BARCODE_UPC_E";
  BarcodeType[BarcodeType["BARCODE_EAN13"] = EscPosPrinterConstants.BARCODE_EAN13] = "BARCODE_EAN13";
  BarcodeType[BarcodeType["BARCODE_JAN13"] = EscPosPrinterConstants.BARCODE_JAN13] = "BARCODE_JAN13";
  BarcodeType[BarcodeType["BARCODE_EAN8"] = EscPosPrinterConstants.BARCODE_EAN8] = "BARCODE_EAN8";
  BarcodeType[BarcodeType["BARCODE_JAN8"] = EscPosPrinterConstants.BARCODE_JAN8] = "BARCODE_JAN8";
  BarcodeType[BarcodeType["BARCODE_CODE39"] = EscPosPrinterConstants.BARCODE_CODE39] = "BARCODE_CODE39";
  BarcodeType[BarcodeType["BARCODE_ITF"] = EscPosPrinterConstants.BARCODE_ITF] = "BARCODE_ITF";
  BarcodeType[BarcodeType["BARCODE_CODABAR"] = EscPosPrinterConstants.BARCODE_CODABAR] = "BARCODE_CODABAR";
  BarcodeType[BarcodeType["BARCODE_CODE93"] = EscPosPrinterConstants.BARCODE_CODE93] = "BARCODE_CODE93";
  BarcodeType[BarcodeType["BARCODE_CODE128"] = EscPosPrinterConstants.BARCODE_CODE128] = "BARCODE_CODE128";
  BarcodeType[BarcodeType["BARCODE_CODE128_AUTO"] = EscPosPrinterConstants.BARCODE_CODE128_AUTO] = "BARCODE_CODE128_AUTO";
  BarcodeType[BarcodeType["BARCODE_GS1_128"] = EscPosPrinterConstants.BARCODE_GS1_128] = "BARCODE_GS1_128";
  BarcodeType[BarcodeType["BARCODE_GS1_DATABAR_OMNIDIRECTIONAL"] = EscPosPrinterConstants.BARCODE_GS1_DATABAR_OMNIDIRECTIONAL] = "BARCODE_GS1_DATABAR_OMNIDIRECTIONAL";
  BarcodeType[BarcodeType["BARCODE_GS1_DATABAR_TRUNCATED"] = EscPosPrinterConstants.BARCODE_GS1_DATABAR_TRUNCATED] = "BARCODE_GS1_DATABAR_TRUNCATED";
  BarcodeType[BarcodeType["BARCODE_GS1_DATABAR_LIMITED"] = EscPosPrinterConstants.BARCODE_GS1_DATABAR_LIMITED] = "BARCODE_GS1_DATABAR_LIMITED";
  BarcodeType[BarcodeType["BARCODE_GS1_DATABAR_EXPANDED"] = EscPosPrinterConstants.BARCODE_GS1_DATABAR_EXPANDED] = "BARCODE_GS1_DATABAR_EXPANDED";
  return BarcodeType;
}({});
export let BarcodeHRI = function (BarcodeHRI) {
  BarcodeHRI[BarcodeHRI["HRI_NONE"] = EscPosPrinterConstants.HRI_NONE] = "HRI_NONE";
  BarcodeHRI[BarcodeHRI["HRI_ABOVE"] = EscPosPrinterConstants.HRI_ABOVE] = "HRI_ABOVE";
  BarcodeHRI[BarcodeHRI["HRI_BELOW"] = EscPosPrinterConstants.HRI_BELOW] = "HRI_BELOW";
  BarcodeHRI[BarcodeHRI["HRI_BOTH"] = EscPosPrinterConstants.HRI_BOTH] = "HRI_BOTH";
  return BarcodeHRI;
}({});
