// react-native-esc-pos-printer/lib/module/printer/constants/enums/image.js
"use strict";

import { EscPosPrinter } from "../../../specs/index.js";
const EscPosPrinterConstants = EscPosPrinter.getConstants();
export let ImageColorModeType = function (ImageColorModeType) {
  ImageColorModeType[ImageColorModeType["MODE_MONO"] = EscPosPrinterConstants.MODE_MONO] = "MODE_MONO";
  ImageColorModeType[ImageColorModeType["MODE_GRAY16"] = EscPosPrinterConstants.MODE_GRAY16] = "MODE_GRAY16";
  ImageColorModeType[ImageColorModeType["MODE_MONO_HIGH_DENSITY"] = EscPosPrinterConstants.MODE_MONO_HIGH_DENSITY] = "MODE_MONO_HIGH_DENSITY";
  return ImageColorModeType;
}({});
export let ImageHalftoneType = function (ImageHalftoneType) {
  ImageHalftoneType[ImageHalftoneType["HALFTONE_DITHER"] = EscPosPrinterConstants.HALFTONE_DITHER] = "HALFTONE_DITHER";
  ImageHalftoneType[ImageHalftoneType["HALFTONE_ERROR_DIFFUSION"] = EscPosPrinterConstants.HALFTONE_ERROR_DIFFUSION] = "HALFTONE_ERROR_DIFFUSION";
  ImageHalftoneType[ImageHalftoneType["HALFTONE_THRESHOLD"] = EscPosPrinterConstants.HALFTONE_THRESHOLD] = "HALFTONE_THRESHOLD";
  return ImageHalftoneType;
}({});
export let ImageCompressType = function (ImageCompressType) {
  ImageCompressType[ImageCompressType["COMPRESS_DEFLATE"] = EscPosPrinterConstants.COMPRESS_DEFLATE] = "COMPRESS_DEFLATE";
  ImageCompressType[ImageCompressType["COMPRESS_NONE"] = EscPosPrinterConstants.COMPRESS_NONE] = "COMPRESS_NONE";
  ImageCompressType[ImageCompressType["COMPRESS_AUTO"] = EscPosPrinterConstants.COMPRESS_AUTO] = "COMPRESS_AUTO";
  return ImageCompressType;
}({});
