// react-native-esc-pos-printer/lib/module/printer/constants/enums/index.js
"use strict";

import { PrinterAddCutType, PrinterErrorResult, PrinterErrorCodeResult, PrinterModelLang, CommonParams, ColorType, TextLangType } from "./common.js";
import { PrinterGetSettingsType, GetPrinterSettingsDensityValues, GetPrinterSettingsPaperWidthValues, GetPrinterSettingsPrintSpeedValues } from "./settings.js";
import { PrinterPaperStatus, PrinterPanelSwitchStatus, PrinterDrawerStatus, PrinterErrorStatus, PrinterAutoRecoverErrorStatus, PrinterBatteryLevelStatus, PrinterRemovalWaitingStatus, PrinterPaperTakenSensorStatus, PrinterUnrecoverErrorStatus } from "./status.js";
import { ImageCompressType, ImageHalftoneType, ImageColorModeType } from "./image.js";
import { BarcodeType, BarcodeHRI } from "./barcode.js";
import { SymbolType, SymbolLevel } from "./symbol.js";
import { FontType, TextAlignType } from "./textStyle.js";
import { PrinterAddPulseDrawerType, PrinterAddPulseTimeType } from "./pulse.js";
export const PrinterConstants = {
  ...PrinterModelLang,
  ...PrinterAddCutType,
  ...PrinterGetSettingsType,
  ...GetPrinterSettingsPaperWidthValues,
  ...GetPrinterSettingsDensityValues,
  ...GetPrinterSettingsPrintSpeedValues,
  ...ColorType,
  ...ImageCompressType,
  ...ImageHalftoneType,
  ...ImageColorModeType,
  ...BarcodeType,
  ...BarcodeHRI,
  ...FontType,
  ...SymbolType,
  ...SymbolLevel,
  ...PrinterAddPulseDrawerType,
  ...PrinterAddPulseTimeType,
  ...TextAlignType,
  ...PrinterPaperStatus,
  ...PrinterPanelSwitchStatus,
  ...PrinterDrawerStatus,
  ...PrinterErrorStatus,
  ...PrinterAutoRecoverErrorStatus,
  ...PrinterBatteryLevelStatus,
  ...PrinterRemovalWaitingStatus,
  ...PrinterPaperTakenSensorStatus,
  ...PrinterUnrecoverErrorStatus,
  ...TextLangType,
  ...CommonParams
};
export { PrinterModelLang, PrinterGetSettingsType, GetPrinterSettingsDensityValues, GetPrinterSettingsPaperWidthValues, GetPrinterSettingsPrintSpeedValues, PrinterAddCutType, PrinterErrorResult, PrinterErrorCodeResult, ColorType, ImageCompressType, ImageHalftoneType, ImageColorModeType, CommonParams, BarcodeType, BarcodeHRI, FontType, SymbolType, SymbolLevel, PrinterAddPulseDrawerType, PrinterAddPulseTimeType, TextAlignType, PrinterPaperStatus, PrinterPanelSwitchStatus, PrinterDrawerStatus, PrinterErrorStatus, PrinterAutoRecoverErrorStatus, PrinterBatteryLevelStatus, PrinterRemovalWaitingStatus, PrinterPaperTakenSensorStatus, PrinterUnrecoverErrorStatus, TextLangType };
