// react-native-esc-pos-printer/lib/module/printer/constants/enums/pulse.js
"use strict";

import { EscPosPrinter } from "../../../specs/index.js";
const EscPosPrinterConstants = EscPosPrinter.getConstants();
export let PrinterAddPulseDrawerType = function (PrinterAddPulseDrawerType) {
  PrinterAddPulseDrawerType[PrinterAddPulseDrawerType["DRAWER_2PIN"] = EscPosPrinterConstants.DRAWER_2PIN] = "DRAWER_2PIN";
  PrinterAddPulseDrawerType[PrinterAddPulseDrawerType["DRAWER_5PIN"] = EscPosPrinterConstants.DRAWER_5PIN] = "DRAWER_5PIN";
  return PrinterAddPulseDrawerType;
}({});
export let PrinterAddPulseTimeType = function (PrinterAddPulseTimeType) {
  PrinterAddPulseTimeType[PrinterAddPulseTimeType["PULSE_100"] = EscPosPrinterConstants.PULSE_100] = "PULSE_100";
  PrinterAddPulseTimeType[PrinterAddPulseTimeType["PULSE_200"] = EscPosPrinterConstants.PULSE_200] = "PULSE_200";
  PrinterAddPulseTimeType[PrinterAddPulseTimeType["PULSE_300"] = EscPosPrinterConstants.PULSE_300] = "PULSE_300";
  PrinterAddPulseTimeType[PrinterAddPulseTimeType["PULSE_400"] = EscPosPrinterConstants.PULSE_400] = "PULSE_400";
  PrinterAddPulseTimeType[PrinterAddPulseTimeType["PULSE_500"] = EscPosPrinterConstants.PULSE_500] = "PULSE_500";
  return PrinterAddPulseTimeType;
}({});
