// react-native-esc-pos-printer/lib/module/printer/constants/enums/symbol.js
"use strict";

import { EscPosPrinter } from "../../../specs/index.js";
const EscPosPrinterConstants = EscPosPrinter.getConstants();
export let SymbolType = function (SymbolType) {
  SymbolType[SymbolType["SYMBOL_PDF417_STANDARD"] = EscPosPrinterConstants.SYMBOL_PDF417_STANDARD] = "SYMBOL_PDF417_STANDARD";
  SymbolType[SymbolType["SYMBOL_PDF417_TRUNCATED"] = EscPosPrinterConstants.SYMBOL_PDF417_TRUNCATED] = "SYMBOL_PDF417_TRUNCATED";
  SymbolType[SymbolType["SYMBOL_QRCODE_MODEL_1"] = EscPosPrinterConstants.SYMBOL_QRCODE_MODEL_1] = "SYMBOL_QRCODE_MODEL_1";
  SymbolType[SymbolType["SYMBOL_QRCODE_MODEL_2"] = EscPosPrinterConstants.SYMBOL_QRCODE_MODEL_2] = "SYMBOL_QRCODE_MODEL_2";
  SymbolType[SymbolType["SYMBOL_QRCODE_MICRO"] = EscPosPrinterConstants.SYMBOL_QRCODE_MICRO] = "SYMBOL_QRCODE_MICRO";
  SymbolType[SymbolType["SYMBOL_MAXICODE_MODE_2"] = EscPosPrinterConstants.SYMBOL_MAXICODE_MODE_2] = "SYMBOL_MAXICODE_MODE_2";
  SymbolType[SymbolType["SYMBOL_MAXICODE_MODE_3"] = EscPosPrinterConstants.SYMBOL_MAXICODE_MODE_3] = "SYMBOL_MAXICODE_MODE_3";
  SymbolType[SymbolType["SYMBOL_MAXICODE_MODE_4"] = EscPosPrinterConstants.SYMBOL_MAXICODE_MODE_4] = "SYMBOL_MAXICODE_MODE_4";
  SymbolType[SymbolType["SYMBOL_MAXICODE_MODE_5"] = EscPosPrinterConstants.SYMBOL_MAXICODE_MODE_5] = "SYMBOL_MAXICODE_MODE_5";
  SymbolType[SymbolType["SYMBOL_MAXICODE_MODE_6"] = EscPosPrinterConstants.SYMBOL_MAXICODE_MODE_6] = "SYMBOL_MAXICODE_MODE_6";
  SymbolType[SymbolType["SYMBOL_GS1_DATABAR_STACKED"] = EscPosPrinterConstants.SYMBOL_GS1_DATABAR_STACKED] = "SYMBOL_GS1_DATABAR_STACKED";
  SymbolType[SymbolType["SYMBOL_GS1_DATABAR_STACKED_OMNIDIRECTIONAL"] = EscPosPrinterConstants.SYMBOL_GS1_DATABAR_STACKED_OMNIDIRECTIONAL] = "SYMBOL_GS1_DATABAR_STACKED_OMNIDIRECTIONAL";
  SymbolType[SymbolType["SYMBOL_GS1_DATABAR_EXPANDED_STACKED"] = EscPosPrinterConstants.SYMBOL_GS1_DATABAR_EXPANDED_STACKED] = "SYMBOL_GS1_DATABAR_EXPANDED_STACKED";
  SymbolType[SymbolType["SYMBOL_AZTECCODE_FULLRANGE"] = EscPosPrinterConstants.SYMBOL_AZTECCODE_FULLRANGE] = "SYMBOL_AZTECCODE_FULLRANGE";
  SymbolType[SymbolType["SYMBOL_AZTECCODE_COMPACT"] = EscPosPrinterConstants.SYMBOL_AZTECCODE_COMPACT] = "SYMBOL_AZTECCODE_COMPACT";
  SymbolType[SymbolType["SYMBOL_DATAMATRIX_SQUARE"] = EscPosPrinterConstants.SYMBOL_DATAMATRIX_SQUARE] = "SYMBOL_DATAMATRIX_SQUARE";
  SymbolType[SymbolType["SYMBOL_DATAMATRIX_RECTANGLE_8"] = EscPosPrinterConstants.SYMBOL_DATAMATRIX_RECTANGLE_8] = "SYMBOL_DATAMATRIX_RECTANGLE_8";
  SymbolType[SymbolType["SYMBOL_DATAMATRIX_RECTANGLE_12"] = EscPosPrinterConstants.SYMBOL_DATAMATRIX_RECTANGLE_12] = "SYMBOL_DATAMATRIX_RECTANGLE_12";
  SymbolType[SymbolType["SYMBOL_DATAMATRIX_RECTANGLE_16"] = EscPosPrinterConstants.SYMBOL_DATAMATRIX_RECTANGLE_16] = "SYMBOL_DATAMATRIX_RECTANGLE_16";
  return SymbolType;
}({});
export let SymbolLevel = function (SymbolLevel) {
  SymbolLevel[SymbolLevel["LEVEL_0"] = EscPosPrinterConstants.LEVEL_0] = "LEVEL_0";
  SymbolLevel[SymbolLevel["LEVEL_1"] = EscPosPrinterConstants.LEVEL_1] = "LEVEL_1";
  SymbolLevel[SymbolLevel["LEVEL_2"] = EscPosPrinterConstants.LEVEL_2] = "LEVEL_2";
  SymbolLevel[SymbolLevel["LEVEL_3"] = EscPosPrinterConstants.LEVEL_3] = "LEVEL_3";
  SymbolLevel[SymbolLevel["LEVEL_4"] = EscPosPrinterConstants.LEVEL_4] = "LEVEL_4";
  SymbolLevel[SymbolLevel["LEVEL_5"] = EscPosPrinterConstants.LEVEL_5] = "LEVEL_5";
  SymbolLevel[SymbolLevel["LEVEL_6"] = EscPosPrinterConstants.LEVEL_6] = "LEVEL_6";
  SymbolLevel[SymbolLevel["LEVEL_7"] = EscPosPrinterConstants.LEVEL_7] = "LEVEL_7";
  SymbolLevel[SymbolLevel["LEVEL_8"] = EscPosPrinterConstants.LEVEL_8] = "LEVEL_8";
  SymbolLevel[SymbolLevel["LEVEL_L"] = EscPosPrinterConstants.LEVEL_L] = "LEVEL_L";
  SymbolLevel[SymbolLevel["LEVEL_M"] = EscPosPrinterConstants.LEVEL_M] = "LEVEL_M";
  SymbolLevel[SymbolLevel["LEVEL_Q"] = EscPosPrinterConstants.LEVEL_Q] = "LEVEL_Q";
  SymbolLevel[SymbolLevel["LEVEL_H"] = EscPosPrinterConstants.LEVEL_H] = "LEVEL_H";
  return SymbolLevel;
}({});
