// react-native-esc-pos-printer/lib/commonjs/printer/constants/enums/image.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ImageHalftoneType = exports.ImageCompressType = exports.ImageColorModeType = void 0;
var _index = require("../../../specs/index.js");
const EscPosPrinterConstants = _index.EscPosPrinter.getConstants();
let ImageColorModeType = exports.ImageColorModeType = function (ImageColorModeType) {
  ImageColorModeType[ImageColorModeType["MODE_MONO"] = EscPosPrinterConstants.MODE_MONO] = "MODE_MONO";
  ImageColorModeType[ImageColorModeType["MODE_GRAY16"] = EscPosPrinterConstants.MODE_GRAY16] = "MODE_GRAY16";
  ImageColorModeType[ImageColorModeType["MODE_MONO_HIGH_DENSITY"] = EscPosPrinterConstants.MODE_MONO_HIGH_DENSITY] = "MODE_MONO_HIGH_DENSITY";
  return ImageColorModeType;
}({});
let ImageHalftoneType = exports.ImageHalftoneType = function (ImageHalftoneType) {
  ImageHalftoneType[ImageHalftoneType["HALFTONE_DITHER"] = EscPosPrinterConstants.HALFTONE_DITHER] = "HALFTONE_DITHER";
  ImageHalftoneType[ImageHalftoneType["HALFTONE_ERROR_DIFFUSION"] = EscPosPrinterConstants.HALFTONE_ERROR_DIFFUSION] = "HALFTONE_ERROR_DIFFUSION";
  ImageHalftoneType[ImageHalftoneType["HALFTONE_THRESHOLD"] = EscPosPrinterConstants.HALFTONE_THRESHOLD] = "HALFTONE_THRESHOLD";
  return ImageHalftoneType;
}({});
let ImageCompressType = exports.ImageCompressType = function (ImageCompressType) {
  ImageCompressType[ImageCompressType["COMPRESS_DEFLATE"] = EscPosPrinterConstants.COMPRESS_DEFLATE] = "COMPRESS_DEFLATE";
  ImageCompressType[ImageCompressType["COMPRESS_NONE"] = EscPosPrinterConstants.COMPRESS_NONE] = "COMPRESS_NONE";
  ImageCompressType[ImageCompressType["COMPRESS_AUTO"] = EscPosPrinterConstants.COMPRESS_AUTO] = "COMPRESS_AUTO";
  return ImageCompressType;
}({});
