// react-native-esc-pos-printer/lib/commonjs/printer/constants/enums/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "BarcodeHRI", {
  enumerable: true,
  get: function () {
    return _barcode.BarcodeHRI;
  }
});
Object.defineProperty(exports, "BarcodeType", {
  enumerable: true,
  get: function () {
    return _barcode.BarcodeType;
  }
});
Object.defineProperty(exports, "ColorType", {
  enumerable: true,
  get: function () {
    return _common.ColorType;
  }
});
Object.defineProperty(exports, "CommonParams", {
  enumerable: true,
  get: function () {
    return _common.CommonParams;
  }
});
Object.defineProperty(exports, "FontType", {
  enumerable: true,
  get: function () {
    return _textStyle.FontType;
  }
});
Object.defineProperty(exports, "GetPrinterSettingsDensityValues", {
  enumerable: true,
  get: function () {
    return _settings.GetPrinterSettingsDensityValues;
  }
});
Object.defineProperty(exports, "GetPrinterSettingsPaperWidthValues", {
  enumerable: true,
  get: function () {
    return _settings.GetPrinterSettingsPaperWidthValues;
  }
});
Object.defineProperty(exports, "GetPrinterSettingsPrintSpeedValues", {
  enumerable: true,
  get: function () {
    return _settings.GetPrinterSettingsPrintSpeedValues;
  }
});
Object.defineProperty(exports, "ImageColorModeType", {
  enumerable: true,
  get: function () {
    return _image.ImageColorModeType;
  }
});
Object.defineProperty(exports, "ImageCompressType", {
  enumerable: true,
  get: function () {
    return _image.ImageCompressType;
  }
});
Object.defineProperty(exports, "ImageHalftoneType", {
  enumerable: true,
  get: function () {
    return _image.ImageHalftoneType;
  }
});
Object.defineProperty(exports, "PrinterAddCutType", {
  enumerable: true,
  get: function () {
    return _common.PrinterAddCutType;
  }
});
Object.defineProperty(exports, "PrinterAddPulseDrawerType", {
  enumerable: true,
  get: function () {
    return _pulse.PrinterAddPulseDrawerType;
  }
});
Object.defineProperty(exports, "PrinterAddPulseTimeType", {
  enumerable: true,
  get: function () {
    return _pulse.PrinterAddPulseTimeType;
  }
});
Object.defineProperty(exports, "PrinterAutoRecoverErrorStatus", {
  enumerable: true,
  get: function () {
    return _status.PrinterAutoRecoverErrorStatus;
  }
});
Object.defineProperty(exports, "PrinterBatteryLevelStatus", {
  enumerable: true,
  get: function () {
    return _status.PrinterBatteryLevelStatus;
  }
});
exports.PrinterConstants = void 0;
Object.defineProperty(exports, "PrinterDrawerStatus", {
  enumerable: true,
  get: function () {
    return _status.PrinterDrawerStatus;
  }
});
Object.defineProperty(exports, "PrinterErrorCodeResult", {
  enumerable: true,
  get: function () {
    return _common.PrinterErrorCodeResult;
  }
});
Object.defineProperty(exports, "PrinterErrorResult", {
  enumerable: true,
  get: function () {
    return _common.PrinterErrorResult;
  }
});
Object.defineProperty(exports, "PrinterErrorStatus", {
  enumerable: true,
  get: function () {
    return _status.PrinterErrorStatus;
  }
});
Object.defineProperty(exports, "PrinterGetSettingsType", {
  enumerable: true,
  get: function () {
    return _settings.PrinterGetSettingsType;
  }
});
Object.defineProperty(exports, "PrinterModelLang", {
  enumerable: true,
  get: function () {
    return _common.PrinterModelLang;
  }
});
Object.defineProperty(exports, "PrinterPanelSwitchStatus", {
  enumerable: true,
  get: function () {
    return _status.PrinterPanelSwitchStatus;
  }
});
Object.defineProperty(exports, "PrinterPaperStatus", {
  enumerable: true,
  get: function () {
    return _status.PrinterPaperStatus;
  }
});
Object.defineProperty(exports, "PrinterPaperTakenSensorStatus", {
  enumerable: true,
  get: function () {
    return _status.PrinterPaperTakenSensorStatus;
  }
});
Object.defineProperty(exports, "PrinterRemovalWaitingStatus", {
  enumerable: true,
  get: function () {
    return _status.PrinterRemovalWaitingStatus;
  }
});
Object.defineProperty(exports, "PrinterUnrecoverErrorStatus", {
  enumerable: true,
  get: function () {
    return _status.PrinterUnrecoverErrorStatus;
  }
});
Object.defineProperty(exports, "SymbolLevel", {
  enumerable: true,
  get: function () {
    return _symbol.SymbolLevel;
  }
});
Object.defineProperty(exports, "SymbolType", {
  enumerable: true,
  get: function () {
    return _symbol.SymbolType;
  }
});
Object.defineProperty(exports, "TextAlignType", {
  enumerable: true,
  get: function () {
    return _textStyle.TextAlignType;
  }
});
Object.defineProperty(exports, "TextLangType", {
  enumerable: true,
  get: function () {
    return _common.TextLangType;
  }
});
var _common = require("./common.js");
var _settings = require("./settings.js");
var _status = require("./status.js");
var _image = require("./image.js");
var _barcode = require("./barcode.js");
var _symbol = require("./symbol.js");
var _textStyle = require("./textStyle.js");
var _pulse = require("./pulse.js");
const PrinterConstants = exports.PrinterConstants = {
  ..._common.PrinterModelLang,
  ..._common.PrinterAddCutType,
  ..._settings.PrinterGetSettingsType,
  ..._settings.GetPrinterSettingsPaperWidthValues,
  ..._settings.GetPrinterSettingsDensityValues,
  ..._settings.GetPrinterSettingsPrintSpeedValues,
  ..._common.ColorType,
  ..._image.ImageCompressType,
  ..._image.ImageHalftoneType,
  ..._image.ImageColorModeType,
  ..._barcode.BarcodeType,
  ..._barcode.BarcodeHRI,
  ..._textStyle.FontType,
  ..._symbol.SymbolType,
  ..._symbol.SymbolLevel,
  ..._pulse.PrinterAddPulseDrawerType,
  ..._pulse.PrinterAddPulseTimeType,
  ..._textStyle.TextAlignType,
  ..._status.PrinterPaperStatus,
  ..._status.PrinterPanelSwitchStatus,
  ..._status.PrinterDrawerStatus,
  ..._status.PrinterErrorStatus,
  ..._status.PrinterAutoRecoverErrorStatus,
  ..._status.PrinterBatteryLevelStatus,
  ..._status.PrinterRemovalWaitingStatus,
  ..._status.PrinterPaperTakenSensorStatus,
  ..._status.PrinterUnrecoverErrorStatus,
  ..._common.TextLangType,
  ..._common.CommonParams
};
