// react-native-esc-pos-printer/lib/commonjs/printer/constants/enums/status.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PrinterUnrecoverErrorStatus = exports.PrinterRemovalWaitingStatus = exports.PrinterPaperTakenSensorStatus = exports.PrinterPaperStatus = exports.PrinterPanelSwitchStatus = exports.PrinterErrorStatus = exports.PrinterDrawerStatus = exports.PrinterBatteryLevelStatus = exports.PrinterAutoRecoverErrorStatus = void 0;
var _index = require("../../../specs/index.js");
const EscPosPrinterConstants = _index.EscPosPrinter.getConstants();
let PrinterPaperStatus = exports.PrinterPaperStatus = function (PrinterPaperStatus) {
  PrinterPaperStatus[PrinterPaperStatus["PAPER_OK"] = EscPosPrinterConstants.PAPER_OK] = "PAPER_OK";
  PrinterPaperStatus[PrinterPaperStatus["PAPER_NEAR_END"] = EscPosPrinterConstants.PAPER_NEAR_END] = "PAPER_NEAR_END";
  PrinterPaperStatus[PrinterPaperStatus["PAPER_EMPTY"] = EscPosPrinterConstants.PAPER_EMPTY] = "PAPER_EMPTY";
  return PrinterPaperStatus;
}({});
let PrinterPanelSwitchStatus = exports.PrinterPanelSwitchStatus = function (PrinterPanelSwitchStatus) {
  PrinterPanelSwitchStatus[PrinterPanelSwitchStatus["SWITCH_ON"] = EscPosPrinterConstants.SWITCH_ON] = "SWITCH_ON";
  PrinterPanelSwitchStatus[PrinterPanelSwitchStatus["SWITCH_OFF"] = EscPosPrinterConstants.SWITCH_OFF] = "SWITCH_OFF";
  return PrinterPanelSwitchStatus;
}({});
let PrinterDrawerStatus = exports.PrinterDrawerStatus = function (PrinterDrawerStatus) {
  PrinterDrawerStatus[PrinterDrawerStatus["DRAWER_HIGH"] = EscPosPrinterConstants.DRAWER_HIGH] = "DRAWER_HIGH";
  PrinterDrawerStatus[PrinterDrawerStatus["DRAWER_LOW"] = EscPosPrinterConstants.DRAWER_LOW] = "DRAWER_LOW";
  return PrinterDrawerStatus;
}({});
let PrinterErrorStatus = exports.PrinterErrorStatus = function (PrinterErrorStatus) {
  PrinterErrorStatus[PrinterErrorStatus["NO_ERR"] = EscPosPrinterConstants.NO_ERR] = "NO_ERR";
  PrinterErrorStatus[PrinterErrorStatus["MECHANICAL_ERR"] = EscPosPrinterConstants.MECHANICAL_ERR] = "MECHANICAL_ERR";
  PrinterErrorStatus[PrinterErrorStatus["AUTOCUTTER_ERR"] = EscPosPrinterConstants.AUTOCUTTER_ERR] = "AUTOCUTTER_ERR";
  PrinterErrorStatus[PrinterErrorStatus["UNRECOVER_ERR"] = EscPosPrinterConstants.UNRECOVER_ERR] = "UNRECOVER_ERR";
  PrinterErrorStatus[PrinterErrorStatus["AUTORECOVER_ERR"] = EscPosPrinterConstants.AUTORECOVER_ERR] = "AUTORECOVER_ERR";
  return PrinterErrorStatus;
}({});
let PrinterAutoRecoverErrorStatus = exports.PrinterAutoRecoverErrorStatus = function (PrinterAutoRecoverErrorStatus) {
  PrinterAutoRecoverErrorStatus[PrinterAutoRecoverErrorStatus["HEAD_OVERHEAT"] = EscPosPrinterConstants.HEAD_OVERHEAT] = "HEAD_OVERHEAT";
  PrinterAutoRecoverErrorStatus[PrinterAutoRecoverErrorStatus["MOTOR_OVERHEAT"] = EscPosPrinterConstants.MOTOR_OVERHEAT] = "MOTOR_OVERHEAT";
  PrinterAutoRecoverErrorStatus[PrinterAutoRecoverErrorStatus["BATTERY_OVERHEAT"] = EscPosPrinterConstants.BATTERY_OVERHEAT] = "BATTERY_OVERHEAT";
  PrinterAutoRecoverErrorStatus[PrinterAutoRecoverErrorStatus["WRONG_PAPER"] = EscPosPrinterConstants.WRONG_PAPER] = "WRONG_PAPER";
  PrinterAutoRecoverErrorStatus[PrinterAutoRecoverErrorStatus["COVER_OPEN"] = EscPosPrinterConstants.COVER_OPEN] = "COVER_OPEN";
  return PrinterAutoRecoverErrorStatus;
}({});
let PrinterBatteryLevelStatus = exports.PrinterBatteryLevelStatus = function (PrinterBatteryLevelStatus) {
  PrinterBatteryLevelStatus[PrinterBatteryLevelStatus["EPOS2_BATTERY_LEVEL_6"] = EscPosPrinterConstants.EPOS2_BATTERY_LEVEL_6] = "EPOS2_BATTERY_LEVEL_6";
  PrinterBatteryLevelStatus[PrinterBatteryLevelStatus["EPOS2_BATTERY_LEVEL_5"] = EscPosPrinterConstants.EPOS2_BATTERY_LEVEL_5] = "EPOS2_BATTERY_LEVEL_5";
  PrinterBatteryLevelStatus[PrinterBatteryLevelStatus["EPOS2_BATTERY_LEVEL_4"] = EscPosPrinterConstants.EPOS2_BATTERY_LEVEL_4] = "EPOS2_BATTERY_LEVEL_4";
  PrinterBatteryLevelStatus[PrinterBatteryLevelStatus["EPOS2_BATTERY_LEVEL_3"] = EscPosPrinterConstants.EPOS2_BATTERY_LEVEL_3] = "EPOS2_BATTERY_LEVEL_3";
  PrinterBatteryLevelStatus[PrinterBatteryLevelStatus["EPOS2_BATTERY_LEVEL_2"] = EscPosPrinterConstants.EPOS2_BATTERY_LEVEL_2] = "EPOS2_BATTERY_LEVEL_2";
  PrinterBatteryLevelStatus[PrinterBatteryLevelStatus["EPOS2_BATTERY_LEVEL_1"] = EscPosPrinterConstants.EPOS2_BATTERY_LEVEL_1] = "EPOS2_BATTERY_LEVEL_1";
  PrinterBatteryLevelStatus[PrinterBatteryLevelStatus["EPOS2_BATTERY_LEVEL_0"] = EscPosPrinterConstants.EPOS2_BATTERY_LEVEL_0] = "EPOS2_BATTERY_LEVEL_0";
  return PrinterBatteryLevelStatus;
}({});
let PrinterRemovalWaitingStatus = exports.PrinterRemovalWaitingStatus = function (PrinterRemovalWaitingStatus) {
  PrinterRemovalWaitingStatus[PrinterRemovalWaitingStatus["REMOVAL_WAIT_PAPER"] = EscPosPrinterConstants.REMOVAL_WAIT_PAPER] = "REMOVAL_WAIT_PAPER";
  PrinterRemovalWaitingStatus[PrinterRemovalWaitingStatus["REMOVAL_WAIT_NONE"] = EscPosPrinterConstants.REMOVAL_WAIT_NONE] = "REMOVAL_WAIT_NONE";
  return PrinterRemovalWaitingStatus;
}({});
let PrinterPaperTakenSensorStatus = exports.PrinterPaperTakenSensorStatus = function (PrinterPaperTakenSensorStatus) {
  PrinterPaperTakenSensorStatus[PrinterPaperTakenSensorStatus["REMOVAL_DETECT_PAPER"] = EscPosPrinterConstants.REMOVAL_DETECT_PAPER] = "REMOVAL_DETECT_PAPER";
  PrinterPaperTakenSensorStatus[PrinterPaperTakenSensorStatus["REMOVAL_DETECT_PAPER_NONE"] = EscPosPrinterConstants.REMOVAL_DETECT_PAPER_NONE] = "REMOVAL_DETECT_PAPER_NONE";
  PrinterPaperTakenSensorStatus[PrinterPaperTakenSensorStatus["REMOVAL_DETECT_UNKNOWN"] = EscPosPrinterConstants.REMOVAL_DETECT_UNKNOWN] = "REMOVAL_DETECT_UNKNOWN";
  return PrinterPaperTakenSensorStatus;
}({});
let PrinterUnrecoverErrorStatus = exports.PrinterUnrecoverErrorStatus = function (PrinterUnrecoverErrorStatus) {
  PrinterUnrecoverErrorStatus[PrinterUnrecoverErrorStatus["HIGH_VOLTAGE_ERR"] = EscPosPrinterConstants.HIGH_VOLTAGE_ERR] = "HIGH_VOLTAGE_ERR";
  PrinterUnrecoverErrorStatus[PrinterUnrecoverErrorStatus["LOW_VOLTAGE_ERR"] = EscPosPrinterConstants.LOW_VOLTAGE_ERR] = "LOW_VOLTAGE_ERR";
  return PrinterUnrecoverErrorStatus;
}({});
