// react-native-esc-pos-printer/lib/commonjs/printer/constants/enums/textStyle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TextAlignType = exports.FontType = void 0;
var _index = require("../../../specs/index.js");
const EscPosPrinterConstants = _index.EscPosPrinter.getConstants();
let FontType = exports.FontType = function (FontType) {
  FontType[FontType["FONT_A"] = EscPosPrinterConstants.FONT_A] = "FONT_A";
  FontType[FontType["FONT_B"] = EscPosPrinterConstants.FONT_B] = "FONT_B";
  FontType[FontType["FONT_C"] = EscPosPrinterConstants.FONT_C] = "FONT_C";
  FontType[FontType["FONT_D"] = EscPosPrinterConstants.FONT_D] = "FONT_D";
  FontType[FontType["FONT_E"] = EscPosPrinterConstants.FONT_E] = "FONT_E";
  return FontType;
}({});
let TextAlignType = exports.TextAlignType = function (TextAlignType) {
  TextAlignType[TextAlignType["ALIGN_LEFT"] = EscPosPrinterConstants.ALIGN_LEFT] = "ALIGN_LEFT";
  TextAlignType[TextAlignType["ALIGN_CENTER"] = EscPosPrinterConstants.ALIGN_CENTER] = "ALIGN_CENTER";
  TextAlignType[TextAlignType["ALIGN_RIGHT"] = EscPosPrinterConstants.ALIGN_RIGHT] = "ALIGN_RIGHT";
  return TextAlignType;
}({});
