// react-native-esc-pos-printer/lib/commonjs/printer/constants/valuesToMessagesMappers.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SendDataPrinterErrorMessageMapping = exports.PrintErrorCodeMessageMapping = exports.InitPrinterErrorMessageMapping = exports.GetPrinterStatusMessageMapping = exports.GetPrinterSettingsPrintSpeendValuesMapping = exports.GetPrinterSettingsPrintDencityValuesMapping = exports.GetPrinterSettingsPaperWidthValuesMapping = exports.DisconnectPrinterErrorMessageMapping = exports.ConnectPrinterErrorMessageMapping = exports.CommonOperationErrorMessageMapping = void 0;
var _index = require("./enums/index.js");
const InitPrinterErrorMessageMapping = exports.InitPrinterErrorMessageMapping = {
  [_index.PrinterErrorResult.ERR_PARAM]: 'An invalid parameter was passed.',
  [_index.PrinterErrorResult.ERR_MEMORY]: 'Memory necessary for processing could not be allocated.',
  [_index.PrinterErrorResult.ERR_UNSUPPORTED]: 'A model name or language not supported was specified.',
  [_index.PrinterErrorResult.ERR_FAILURE]: 'An unknown error occurred.'
};
const ConnectPrinterErrorMessageMapping = exports.ConnectPrinterErrorMessageMapping = {
  [_index.PrinterErrorResult.ERR_PARAM]: 'An invalid parameter was passed.',
  [_index.PrinterErrorResult.ERR_CONNECT]: 'Failed to open the device.',
  [_index.PrinterErrorResult.ERR_TIMEOUT]: 'Failed to communicate with the devices within the specified time.',
  [_index.PrinterErrorResult.ERR_ILLEGAL]: 'Tried to start communication with a printer with which communication had been already established. Tried to start communication with a printer during reconnection process.',
  [_index.PrinterErrorResult.ERR_MEMORY]: 'Memory necessary for processing could not be allocated.',
  [_index.PrinterErrorResult.ERR_FAILURE]: 'An unknown error occurred.',
  [_index.PrinterErrorResult.ERR_PROCESSING]: 'Could not run the process.',
  [_index.PrinterErrorResult.ERR_NOT_FOUND]: 'The device could not be found.',
  [_index.PrinterErrorResult.ERR_IN_USE]: 'The device was in use.',
  [_index.PrinterErrorResult.ERR_TYPE_INVALID]: 'The device type is different.',
  [_index.PrinterErrorResult.ERR_RECOVERY_FAILURE]: 'Failed to recover the printer.'
};
const DisconnectPrinterErrorMessageMapping = exports.DisconnectPrinterErrorMessageMapping = {
  [_index.PrinterErrorResult.ERR_ILLEGAL]: 'Tried to end communication where it had not been established.',
  [_index.PrinterErrorResult.ERR_MEMORY]: 'Necessary memory could not be allocated.',
  [_index.PrinterErrorResult.ERR_FAILURE]: 'An unknown error occurred.',
  [_index.PrinterErrorResult.ERR_PROCESSING]: 'Could not run the process.',
  [_index.PrinterErrorResult.ERR_DISCONNECT]: 'Failed to disconnect the device.T ried to terminate communication with a printer during reconnection process.',
  [_index.PrinterErrorResult.ERR_INIT]: 'Printer is not initialized. Please call init() first.'
};
const CommonOperationErrorMessageMapping = exports.CommonOperationErrorMessageMapping = {
  [_index.PrinterErrorResult.ERR_PARAM]: 'An invalid parameter was passed.',
  [_index.PrinterErrorResult.ERR_MEMORY]: 'Memory necessary for processing could not be allocated.',
  [_index.PrinterErrorResult.ERR_FAILURE]: 'An unknown error occurred.',
  [_index.PrinterErrorResult.ERR_INIT]: 'Printer is not initialized. Please call init() first.'
};
const SendDataPrinterErrorMessageMapping = exports.SendDataPrinterErrorMessageMapping = {
  [_index.PrinterErrorResult.ERR_PARAM]: 'An invalid parameter was passed.',
  [_index.PrinterErrorResult.ERR_MEMORY]: 'Memory necessary for processing could not be allocated.',
  [_index.PrinterErrorResult.ERR_FAILURE]: 'An unknown error occurred.',
  [_index.PrinterErrorResult.ERR_PROCESSING]: 'Could not run the process.',
  [_index.PrinterErrorResult.ERR_ILLEGAL]: 'The control commands have not been buffered. This API was called while no communication had been started.',
  [_index.PrinterErrorResult.ERR_INIT]: 'Printer is not initialized. Please call init() first.'
};
const PrintErrorCodeMessageMapping = exports.PrintErrorCodeMessageMapping = {
  [_index.PrinterErrorCodeResult.CODE_ERR_AUTORECOVER]: 'Automatic recovery error occurred.',
  [_index.PrinterErrorCodeResult.CODE_ERR_COVER_OPEN]: 'Cover open error occurred.',
  [_index.PrinterErrorCodeResult.CODE_ERR_CUTTER]: 'Auto cutter error occurred.',
  [_index.PrinterErrorCodeResult.CODE_ERR_MECHANICAL]: 'Mechanical error occurred.',
  [_index.PrinterErrorCodeResult.CODE_ERR_EMPTY]: 'No paper is left in the roll paper end detector.',
  [_index.PrinterErrorCodeResult.CODE_ERR_UNRECOVERABLE]: 'Unrecoverable error occurred.',
  [_index.PrinterErrorCodeResult.CODE_ERR_FAILURE]: 'Error exists in the requested document syntax.',
  [_index.PrinterErrorCodeResult.CODE_ERR_NOT_FOUND]: 'Printer specified by the device ID does not exist.',
  [_index.PrinterErrorCodeResult.CODE_ERR_SYSTEM]: 'Error occurred with the printing system.',
  [_index.PrinterErrorCodeResult.CODE_ERR_PORT]: 'Error was detected with the communication port.',
  [_index.PrinterErrorCodeResult.CODE_ERR_TIMEOUT]: 'Print timeout occurred.',
  [_index.PrinterErrorCodeResult.CODE_ERR_JOB_NOT_FOUND]: 'Specified print job ID does not exist.',
  [_index.PrinterErrorCodeResult.CODE_ERR_SPOOLER]: 'Print queue is full.',
  [_index.PrinterErrorCodeResult.CODE_ERR_BATTERY_LOW]: 'Battery has run out.',
  [_index.PrinterErrorCodeResult.CODE_ERR_TOO_MANY_REQUESTS]: 'The number of print jobs sent to the printer has exceeded the allowable limit.',
  [_index.PrinterErrorCodeResult.CODE_ERR_REQUEST_ENTITY_TOO_LARGE]: 'The size of the print job data exceeds the capacity of the printer.',
  [_index.PrinterErrorCodeResult.CODE_ERR_WAIT_REMOVAL]: 'Print command sent while waiting for paper removal.',
  [_index.PrinterErrorCodeResult.CODE_PRINTING]: 'Printing.',
  [_index.PrinterErrorCodeResult.CODE_ERR_MEMORY]: 'Sufficient memory required for processing could not be allocated.',
  [_index.PrinterErrorCodeResult.CODE_ERR_PROCESSING]: 'Could not run the process.',
  [_index.PrinterErrorCodeResult.CODE_ERR_ILLEGAL]: 'This API was called while no communication had been started.',
  [_index.PrinterErrorCodeResult.CODE_ERR_DEVICE_BUSY]: 'Could not run the process because the printer is operating.'
};
const GetPrinterSettingsPaperWidthValuesMapping = exports.GetPrinterSettingsPaperWidthValuesMapping = {
  [_index.GetPrinterSettingsPaperWidthValues.PRINTER_SETTING_PAPERWIDTH58_0]: 58,
  [_index.GetPrinterSettingsPaperWidthValues.PRINTER_SETTING_PAPERWIDTH60_0]: 60,
  [_index.GetPrinterSettingsPaperWidthValues.PRINTER_SETTING_PAPERWIDTH70_0]: 70,
  [_index.GetPrinterSettingsPaperWidthValues.PRINTER_SETTING_PAPERWIDTH76_0]: 76,
  [_index.GetPrinterSettingsPaperWidthValues.PRINTER_SETTING_PAPERWIDTH80_0]: 80
};
const GetPrinterSettingsPrintDencityValuesMapping = exports.GetPrinterSettingsPrintDencityValuesMapping = {
  [_index.GetPrinterSettingsDensityValues.PRINTER_SETTING_PRINTDENSITYDIP]: 0,
  [_index.GetPrinterSettingsDensityValues.PRINTER_SETTING_PRINTDENSITY70]: 70,
  [_index.GetPrinterSettingsDensityValues.PRINTER_SETTING_PRINTDENSITY75]: 75,
  [_index.GetPrinterSettingsDensityValues.PRINTER_SETTING_PRINTDENSITY80]: 80,
  [_index.GetPrinterSettingsDensityValues.PRINTER_SETTING_PRINTDENSITY85]: 85,
  [_index.GetPrinterSettingsDensityValues.PRINTER_SETTING_PRINTDENSITY90]: 90,
  [_index.GetPrinterSettingsDensityValues.PRINTER_SETTING_PRINTDENSITY95]: 95,
  [_index.GetPrinterSettingsDensityValues.PRINTER_SETTING_PRINTDENSITY100]: 100,
  [_index.GetPrinterSettingsDensityValues.PRINTER_SETTING_PRINTDENSITY105]: 105,
  [_index.GetPrinterSettingsDensityValues.PRINTER_SETTING_PRINTDENSITY110]: 110,
  [_index.GetPrinterSettingsDensityValues.PRINTER_SETTING_PRINTDENSITY115]: 115,
  [_index.GetPrinterSettingsDensityValues.PRINTER_SETTING_PRINTDENSITY120]: 120,
  [_index.GetPrinterSettingsDensityValues.PRINTER_SETTING_PRINTDENSITY125]: 125,
  [_index.GetPrinterSettingsDensityValues.PRINTER_SETTING_PRINTDENSITY130]: 130
};
const GetPrinterSettingsPrintSpeendValuesMapping = exports.GetPrinterSettingsPrintSpeendValuesMapping = {
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED1]: 1,
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED2]: 2,
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED3]: 3,
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED4]: 4,
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED5]: 5,
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED6]: 6,
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED7]: 7,
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED8]: 8,
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED9]: 9,
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED10]: 10,
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED11]: 11,
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED12]: 12,
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED13]: 13,
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED14]: 14,
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED15]: 15,
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED16]: 16,
  [_index.GetPrinterSettingsPrintSpeedValues.PRINTER_SETTING_PRINTSPEED17]: 17
};
const GetPrinterStatusMessageMapping = exports.GetPrinterStatusMessageMapping = {
  connection: {
    [_index.CommonParams.TRUE]: 'Connected',
    [_index.CommonParams.FALSE]: 'Status is unknown.'
  },
  online: {
    [_index.CommonParams.TRUE]: 'Online',
    [_index.CommonParams.FALSE]: 'Offline',
    [_index.CommonParams.UNKNOWN]: 'Status is unknown.'
  },
  coverOpen: {
    [_index.CommonParams.TRUE]: 'Cover is open.',
    [_index.CommonParams.FALSE]: 'Cover is closed.',
    [_index.CommonParams.UNKNOWN]: 'Status is unknown.'
  },
  paper: {
    [_index.PrinterPaperStatus.PAPER_OK]: 'Paper remains.',
    [_index.PrinterPaperStatus.PAPER_NEAR_END]: 'Paper is running out.',
    [_index.PrinterPaperStatus.PAPER_EMPTY]: 'Paper has run out.',
    [_index.CommonParams.UNKNOWN]: 'Status is unknown.'
  },
  paperFeed: {
    [_index.CommonParams.TRUE]: 'Paper feed in progress',
    [_index.CommonParams.FALSE]: 'Stopped',
    [_index.CommonParams.UNKNOWN]: 'Status is unknown.'
  },
  panelSwitch: {
    [_index.PrinterPanelSwitchStatus.SWITCH_ON]: 'Pressed',
    [_index.PrinterPanelSwitchStatus.SWITCH_OFF]: 'Not pressed',
    [_index.CommonParams.UNKNOWN]: 'Status is unknown.'
  },
  drawer: {
    [_index.PrinterDrawerStatus.DRAWER_HIGH]: 'High',
    [_index.PrinterDrawerStatus.DRAWER_LOW]: 'Low',
    [_index.CommonParams.UNKNOWN]: 'Status is unknown.'
  },
  errorStatus: {
    [_index.PrinterErrorStatus.NO_ERR]: 'Normal',
    [_index.PrinterErrorStatus.MECHANICAL_ERR]: 'Mechanical error occurred.',
    [_index.PrinterErrorStatus.AUTOCUTTER_ERR]: 'Auto cutter error occurred.',
    [_index.PrinterErrorStatus.UNRECOVER_ERR]: 'Unrecoverable error occurred.',
    [_index.PrinterErrorStatus.AUTORECOVER_ERR]: 'Automatic recovery error occurred.',
    [_index.CommonParams.UNKNOWN]: 'Status is unknown.'
  },
  autoRecoverError: {
    [_index.PrinterAutoRecoverErrorStatus.HEAD_OVERHEAT]: 'Head overheat error',
    [_index.PrinterAutoRecoverErrorStatus.MOTOR_OVERHEAT]: 'Motor driver IC overheat error',
    [_index.PrinterAutoRecoverErrorStatus.BATTERY_OVERHEAT]: 'Battery overheat error',
    [_index.PrinterAutoRecoverErrorStatus.WRONG_PAPER]: 'Paper error',
    [_index.PrinterAutoRecoverErrorStatus.COVER_OPEN]: 'Cover is open.',
    [_index.CommonParams.UNKNOWN]: 'Status is unknown.'
  },
  buzzer: {
    [_index.CommonParams.TRUE]: 'Sounding',
    [_index.CommonParams.FALSE]: 'Stopped',
    [_index.CommonParams.UNKNOWN]: 'Status is unknown.'
  },
  adapter: {
    [_index.CommonParams.TRUE]: 'Connected',
    [_index.CommonParams.FALSE]: 'Disconnected',
    [_index.CommonParams.UNKNOWN]: 'Status is unknown.'
  },
  batteryLevel: {
    [_index.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_6]: 'Remaining battery capacity 6',
    [_index.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_5]: 'Remaining battery capacity 5',
    [_index.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_4]: 'Remaining battery capacity 4',
    [_index.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_3]: 'Remaining battery capacity 3',
    [_index.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_2]: 'Remaining battery capacity 2',
    [_index.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_1]: 'Remaining battery capacity 1',
    [_index.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_0]: 'Remaining battery capacity 0',
    [_index.CommonParams.UNKNOWN]: 'Status is unknown.'
  },
  removalWaiting: {
    [_index.PrinterRemovalWaitingStatus.REMOVAL_WAIT_PAPER]: 'Waiting for paper removal',
    [_index.PrinterRemovalWaitingStatus.REMOVAL_WAIT_NONE]: 'Not waiting for removal.',
    [_index.CommonParams.UNKNOWN]: 'Status is unknown.'
  },
  paperTakenSensor: {
    [_index.PrinterPaperTakenSensorStatus.REMOVAL_DETECT_PAPER]: 'The paper removal sensor is detecting paper',
    [_index.PrinterPaperTakenSensorStatus.REMOVAL_DETECT_PAPER_NONE]: 'The paper removal sensor is not detecting paper.',
    [_index.PrinterPaperTakenSensorStatus.REMOVAL_DETECT_UNKNOWN]: 'A state that is not detectable by the paper removal sensor',
    [_index.CommonParams.UNKNOWN]: 'Status is unknown.'
  },
  unrecoverError: {
    [_index.PrinterUnrecoverErrorStatus.HIGH_VOLTAGE_ERR]: 'High voltage error',
    [_index.PrinterUnrecoverErrorStatus.LOW_VOLTAGE_ERR]: 'Low voltage error',
    [_index.CommonParams.UNKNOWN]: 'Status is unknown.'
  }
};
