// react-native-esc-pos-printer/lib/commonjs/printer/constants/valuesToKeysMappers.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PrinterStatusMapping = exports.PrinterErrorStatusMapping = exports.PrinterErrorCodeStatusMapping = void 0;
var _index = require("../../core/utils/index.js");
var _index2 = require("./enums/index.js");
const PrinterErrorStatusMapping = exports.PrinterErrorStatusMapping = (0, _index.remapConstants)(_index2.PrinterErrorResult);
const PrinterErrorCodeStatusMapping = exports.PrinterErrorCodeStatusMapping = (0, _index.remapConstants)(_index2.PrinterErrorCodeResult);
const PrinterStatusMapping = exports.PrinterStatusMapping = {
  connection: {
    [_index2.CommonParams.TRUE]: _index2.CommonParams[_index2.CommonParams.TRUE],
    [_index2.CommonParams.FALSE]: _index2.CommonParams[_index2.CommonParams.FALSE]
  },
  online: {
    [_index2.CommonParams.TRUE]: _index2.CommonParams[_index2.CommonParams.TRUE],
    [_index2.CommonParams.FALSE]: _index2.CommonParams[_index2.CommonParams.FALSE],
    [_index2.CommonParams.UNKNOWN]: _index2.CommonParams[_index2.CommonParams.UNKNOWN]
  },
  coverOpen: {
    [_index2.CommonParams.TRUE]: _index2.CommonParams[_index2.CommonParams.TRUE],
    [_index2.CommonParams.FALSE]: _index2.CommonParams[_index2.CommonParams.FALSE],
    [_index2.CommonParams.UNKNOWN]: _index2.CommonParams[_index2.CommonParams.UNKNOWN]
  },
  paper: {
    [_index2.PrinterPaperStatus.PAPER_OK]: _index2.PrinterPaperStatus[_index2.PrinterPaperStatus.PAPER_OK],
    [_index2.PrinterPaperStatus.PAPER_NEAR_END]: _index2.PrinterPaperStatus[_index2.PrinterPaperStatus.PAPER_NEAR_END],
    [_index2.PrinterPaperStatus.PAPER_EMPTY]: _index2.PrinterPaperStatus[_index2.PrinterPaperStatus.PAPER_EMPTY],
    [_index2.CommonParams.UNKNOWN]: _index2.CommonParams[_index2.CommonParams.UNKNOWN]
  },
  paperFeed: {
    [_index2.CommonParams.TRUE]: _index2.CommonParams[_index2.CommonParams.TRUE],
    [_index2.CommonParams.FALSE]: _index2.CommonParams[_index2.CommonParams.FALSE],
    [_index2.CommonParams.UNKNOWN]: _index2.CommonParams[_index2.CommonParams.UNKNOWN]
  },
  panelSwitch: {
    [_index2.PrinterPanelSwitchStatus.SWITCH_ON]: _index2.PrinterPanelSwitchStatus[_index2.PrinterPanelSwitchStatus.SWITCH_ON],
    [_index2.PrinterPanelSwitchStatus.SWITCH_OFF]: _index2.PrinterPanelSwitchStatus[_index2.PrinterPanelSwitchStatus.SWITCH_OFF],
    [_index2.CommonParams.UNKNOWN]: _index2.CommonParams[_index2.CommonParams.UNKNOWN]
  },
  drawer: {
    [_index2.PrinterDrawerStatus.DRAWER_HIGH]: _index2.PrinterDrawerStatus[_index2.PrinterDrawerStatus.DRAWER_HIGH],
    [_index2.PrinterDrawerStatus.DRAWER_LOW]: _index2.PrinterDrawerStatus[_index2.PrinterDrawerStatus.DRAWER_LOW],
    [_index2.CommonParams.UNKNOWN]: _index2.CommonParams[_index2.CommonParams.UNKNOWN]
  },
  errorStatus: {
    [_index2.PrinterErrorStatus.NO_ERR]: _index2.PrinterErrorStatus[_index2.PrinterErrorStatus.NO_ERR],
    [_index2.PrinterErrorStatus.MECHANICAL_ERR]: _index2.PrinterErrorStatus[_index2.PrinterErrorStatus.MECHANICAL_ERR],
    [_index2.PrinterErrorStatus.AUTOCUTTER_ERR]: _index2.PrinterErrorStatus[_index2.PrinterErrorStatus.AUTOCUTTER_ERR],
    [_index2.PrinterErrorStatus.UNRECOVER_ERR]: _index2.PrinterErrorStatus[_index2.PrinterErrorStatus.UNRECOVER_ERR],
    [_index2.PrinterErrorStatus.AUTORECOVER_ERR]: _index2.PrinterErrorStatus[_index2.PrinterErrorStatus.AUTORECOVER_ERR],
    [_index2.CommonParams.UNKNOWN]: _index2.CommonParams[_index2.CommonParams.UNKNOWN]
  },
  autoRecoverError: {
    [_index2.PrinterAutoRecoverErrorStatus.HEAD_OVERHEAT]: _index2.PrinterAutoRecoverErrorStatus[_index2.PrinterAutoRecoverErrorStatus.HEAD_OVERHEAT],
    [_index2.PrinterAutoRecoverErrorStatus.MOTOR_OVERHEAT]: _index2.PrinterAutoRecoverErrorStatus[_index2.PrinterAutoRecoverErrorStatus.MOTOR_OVERHEAT],
    [_index2.PrinterAutoRecoverErrorStatus.BATTERY_OVERHEAT]: _index2.PrinterAutoRecoverErrorStatus[_index2.PrinterAutoRecoverErrorStatus.BATTERY_OVERHEAT],
    [_index2.PrinterAutoRecoverErrorStatus.WRONG_PAPER]: _index2.PrinterAutoRecoverErrorStatus[_index2.PrinterAutoRecoverErrorStatus.WRONG_PAPER],
    [_index2.PrinterAutoRecoverErrorStatus.COVER_OPEN]: _index2.PrinterAutoRecoverErrorStatus[_index2.PrinterAutoRecoverErrorStatus.COVER_OPEN],
    [_index2.CommonParams.UNKNOWN]: _index2.CommonParams[_index2.CommonParams.UNKNOWN]
  },
  buzzer: {
    [_index2.CommonParams.TRUE]: _index2.CommonParams[_index2.CommonParams.TRUE],
    [_index2.CommonParams.FALSE]: _index2.CommonParams[_index2.CommonParams.FALSE],
    [_index2.CommonParams.UNKNOWN]: _index2.CommonParams[_index2.CommonParams.UNKNOWN]
  },
  adapter: {
    [_index2.CommonParams.TRUE]: _index2.CommonParams[_index2.CommonParams.TRUE],
    [_index2.CommonParams.FALSE]: _index2.CommonParams[_index2.CommonParams.FALSE],
    [_index2.CommonParams.UNKNOWN]: _index2.CommonParams[_index2.CommonParams.UNKNOWN]
  },
  batteryLevel: {
    [_index2.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_6]: _index2.PrinterBatteryLevelStatus[_index2.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_6],
    [_index2.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_5]: _index2.PrinterBatteryLevelStatus[_index2.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_5],
    [_index2.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_4]: _index2.PrinterBatteryLevelStatus[_index2.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_4],
    [_index2.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_3]: _index2.PrinterBatteryLevelStatus[_index2.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_3],
    [_index2.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_2]: _index2.PrinterBatteryLevelStatus[_index2.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_2],
    [_index2.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_1]: _index2.PrinterBatteryLevelStatus[_index2.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_1],
    [_index2.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_0]: _index2.PrinterBatteryLevelStatus[_index2.PrinterBatteryLevelStatus.EPOS2_BATTERY_LEVEL_0],
    [_index2.CommonParams.UNKNOWN]: _index2.CommonParams[_index2.CommonParams.UNKNOWN]
  },
  removalWaiting: {
    [_index2.PrinterRemovalWaitingStatus.REMOVAL_WAIT_PAPER]: _index2.PrinterRemovalWaitingStatus[_index2.PrinterRemovalWaitingStatus.REMOVAL_WAIT_PAPER],
    [_index2.PrinterRemovalWaitingStatus.REMOVAL_WAIT_NONE]: _index2.PrinterRemovalWaitingStatus[_index2.PrinterRemovalWaitingStatus.REMOVAL_WAIT_NONE],
    [_index2.CommonParams.UNKNOWN]: _index2.CommonParams[_index2.CommonParams.UNKNOWN]
  },
  paperTakenSensor: {
    [_index2.PrinterPaperTakenSensorStatus.REMOVAL_DETECT_PAPER]: _index2.PrinterPaperTakenSensorStatus[_index2.PrinterPaperTakenSensorStatus.REMOVAL_DETECT_PAPER],
    [_index2.PrinterPaperTakenSensorStatus.REMOVAL_DETECT_PAPER_NONE]: _index2.PrinterPaperTakenSensorStatus[_index2.PrinterPaperTakenSensorStatus.REMOVAL_DETECT_PAPER_NONE],
    [_index2.PrinterPaperTakenSensorStatus.REMOVAL_DETECT_UNKNOWN]: _index2.PrinterPaperTakenSensorStatus[_index2.PrinterPaperTakenSensorStatus.REMOVAL_DETECT_UNKNOWN],
    [_index2.CommonParams.UNKNOWN]: _index2.CommonParams[_index2.CommonParams.UNKNOWN]
  },
  unrecoverError: {
    [_index2.PrinterUnrecoverErrorStatus.HIGH_VOLTAGE_ERR]: _index2.PrinterUnrecoverErrorStatus[_index2.PrinterUnrecoverErrorStatus.HIGH_VOLTAGE_ERR],
    [_index2.PrinterUnrecoverErrorStatus.LOW_VOLTAGE_ERR]: _index2.PrinterUnrecoverErrorStatus[_index2.PrinterUnrecoverErrorStatus.LOW_VOLTAGE_ERR],
    [_index2.CommonParams.UNKNOWN]: _index2.CommonParams[_index2.CommonParams.UNKNOWN]
  }
};
